### ISSUE TITLE
Remove bower related code

### ISSUE THREAD

**Issue by mehulkar** [ember-cli/ember-cli#9110]
There's still Bower related code all over the codebase that can be removed.

**Comment by rwjblue** [comment]
I don’t think we can remove it before ember-cli@4

**Comment by stefanpenner** [comment]
In theory, we may be able to isolate it further, so it's easy to shed? 
Concretely, if there are no deprecations yet, we may want to consider adding them.

**Comment by mehulkar** [comment]
Wasn't bower removed from the blueprints way back in 2.something? I think it would be safe to interpret that as a "no longer supported" and the breaking change being part of 3.x?

**Comment by rwjblue** [comment]
@mehulkar

> Wasn't bower removed from the blueprints way back in 2.something? I think it would be safe to interpret that as a "no longer supported" and the breaking change being part of 3.x?

Ember was consumed from Bower up until ember-source@2.11.0, and eventually (after we started publishing `ember-source` to npm) we stopped emitting a `bower.json` in the blueprint in `ember-cli@2.13.0`. Removing `bower.json` from the default blueprint is **absolutely** not the same as "no longer supported", in fact the migration path that we have followed was specifically geared towards making it an easy / smooth transition for folks.

At this point, we still have a number of internal APIs that inherently support usage with Bower (e.g. `Project.prototype.bowerDependencies`, `Blueprint.prototype.addBowerPackagesToProject`, `app.import` support, ). Before we can remove Bower support, we need to deprecate these APIs.
